wallet/src/de/schildbach/wallet/ui/invite/InviteFilterSelectionDialog.kt

```diff
@@ -41,51 +40,45 @@ class InviteFilterSelectionDialog(private val owner: ViewModelStoreOwner) : Base
+    private val binding get() = _binding!!
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
-        return inflater.inflate(R.layout.dialog_invite_filter, container, false)
+    private val sharedViewModel: InvitesHistoryFilterViewModel by viewModels({ owner })
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = DialogInviteFilterBinding.inflate(inflater)
+        return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        val array = view.context.resources.getStringArray(R.array.invite_filter);
-        firstItem.text = array[0]
-        secondItem.text = array[1]
-        thirdItem.text = array[2]
+        val array = view.context.resources.getStringArray(R.array.invite_filter)
+        binding.firstItem.text = array[0]
+        binding.secondItem.text = array[1]
+        binding.thirdItem.text = array[2]
+
         // take care of actions here
-        view.apply {
-            first.setOnClickListener {
-                sharedViewModel.filterBy.call(InvitesHistoryViewModel.Filter.ALL)
-                dismiss()
-            }
-            second.setOnClickListener {
-                sharedViewModel.filterBy.call(InvitesHistoryViewModel.Filter.PENDING)
-                dismiss()
-            }
-            third.setOnClickListener {
-                sharedViewModel.filterBy.call(InvitesHistoryViewModel.Filter.CLAIMED)
-                dismiss()
-            }
+        binding.first.setOnClickListener {
+            sharedViewModel.setFilter(InvitesHistoryViewModel.Filter.ALL)
+            dismiss()
         }
-
-        dialog?.setOnShowListener { dialog ->
-            // apply wrap_content height
-            val d = dialog as BottomSheetDialog
-            val bottomSheet = d.findViewById<FrameLayout>(R.id.design_bottom_sheet)
-            val coordinatorLayout = bottomSheet!!.parent as CoordinatorLayout
-            val bottomSheetBehavior = BottomSheetBehavior.from(bottomSheet)
-            bottomSheetBehavior.peekHeight = bottomSheet.height
-            coordinatorLayout.parent.requestLayout()
+        binding.second.setOnClickListener {
+            sharedViewModel.setFilter(InvitesHistoryViewModel.Filter.PENDING)
+            dismiss()
+        }
+        binding.third.setOnClickListener {
+            sharedViewModel.setFilter(InvitesHistoryViewModel.Filter.CLAIMED)
+            dismiss()
         }
     }
 
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
-        sharedViewModel = activity?.run {
-            ViewModelProvider(owner)[InvitesHistoryFilterViewModel::class.java]
-        } ?: throw IllegalStateException("Invalid Activity")
```

Comment:
Replaced `ViewModelProvider `with `viewModels({ owner })` and removed this method since it doesn't seem to be needed anymore? Let me know if I'm missing something

Reply:
We didn't know about `viewModels({ owner })`